tests/avifyuv.c

```diff
@@ -106,140 +113,175 @@ int main(int argc, char * argv[])
-                                                                    AVIF_MATRIX_COEFFICIENTS_YCGCO };
-                const int matrixCoeffsCount = (int)(sizeof(matrixCoeffsList) / sizeof(matrixCoeffsList[0]));
-
-                for (int matrixCoeffsIndex = 0; matrixCoeffsIndex < matrixCoeffsCount; ++matrixCoeffsIndex) {
-                    avifMatrixCoefficients matrixCoeffs = matrixCoeffsList[matrixCoeffsIndex];
-
-                    int dim = 1 << rgbDepth;
-                    int maxDrift = 0;
-
-                    avifImage * image = avifImageCreate(dim, dim, yuvDepth, AVIF_PIXEL_FORMAT_YUV444);
-                    image->colorPrimaries = AVIF_COLOR_PRIMARIES_BT709;
-                    image->transferCharacteristics = AVIF_TRANSFER_CHARACTERISTICS_SRGB;
-                    image->matrixCoefficients = matrixCoeffs;
-                    image->yuvRange = AVIF_RANGE_FULL;
-                    avifImageAllocatePlanes(image, AVIF_PLANES_YUV);
-
-                    avifRGBImage srcRGB;
-                    avifRGBImageSetDefaults(&srcRGB, image);
-                    srcRGB.format = AVIF_RGB_FORMAT_RGB;
-                    srcRGB.depth = rgbDepth;
-                    avifRGBImageAllocatePixels(&srcRGB);
-
-                    avifRGBImage dstRGB;
-                    avifRGBImageSetDefaults(&dstRGB, image);
-                    dstRGB.format = AVIF_RGB_FORMAT_RGB;
-                    dstRGB.depth = rgbDepth;
-                    avifRGBImageAllocatePixels(&dstRGB);
-
-                    uint64_t driftPixelCounts[MAX_DRIFT];
-                    for (int i = 0; i < MAX_DRIFT; ++i) {
-                        driftPixelCounts[i] = 0;
-                    }
-
-                    for (int r = 0; r < dim; ++r) {
-                        if (verbose) {
-                            printf("[%4d/%4d] RGB depth: %d, YUV depth: %d, matrixCoeffs: %d\r", r + 1, dim, rgbDepth, yuvDepth, matrixCoeffs);
+                const avifCICP CICPsList[] = {
+                    { AVIF_COLOR_PRIMARIES_BT709, AVIF_TRANSFER_CHARACTERISTICS_SRGB, AVIF_MATRIX_COEFFICIENTS_BT709 },
+                    { AVIF_COLOR_PRIMARIES_BT709, AVIF_TRANSFER_CHARACTERISTICS_SRGB, AVIF_MATRIX_COEFFICIENTS_BT601 },
+                    { AVIF_COLOR_PRIMARIES_BT709, AVIF_TRANSFER_CHARACTERISTICS_SRGB, AVIF_MATRIX_COEFFICIENTS_BT2020_NCL },
+                    { AVIF_COLOR_PRIMARIES_BT709, AVIF_TRANSFER_CHARACTERISTICS_SRGB, AVIF_MATRIX_COEFFICIENTS_IDENTITY },
+                    { AVIF_COLOR_PRIMARIES_BT709, AVIF_TRANSFER_CHARACTERISTICS_SRGB, AVIF_MATRIX_COEFFICIENTS_YCGCO },
+                    { AVIF_COLOR_PRIMARIES_SMPTE432, AVIF_TRANSFER_CHARACTERISTICS_SRGB, AVIF_MATRIX_COEFFICIENTS_CHROMA_DERIVED_NCL },
+                };
+                const int CICPsCount = (int)(sizeof(CICPsList) / sizeof(CICPsList[0]));
+
+                for (int CICPsIndex = 0; CICPsIndex < CICPsCount; ++CICPsIndex) {
+                    avifCICP cicp = CICPsList[CICPsIndex];
```

Comment:
Nit: This variable can be declared as a pointer to avoid copying the avifCICP struct:

                    const avifCICP *cicp = &cicpList[cicpIndex];